File: 02-eda-1/bar-histogram.py
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("movies.csv")
df = df.dropna()
# Map string to int
df["VOTES"] = df["VOTES"].str.replace(",", "").astype(int)

plt.hist(df["VOTES"], bins=10, edgecolor="black")
plt.xlabel("Votes")
plt.ylabel("Frequency")
plt.title("Histogram of Votes")
plt.show()

# Reduce data to 10 records for better plot
df = df[:10]
plt.bar(df["MOVIES"], df["VOTES"], edgecolor="black")
plt.xlabel("Movies")
plt.ylabel("Votes")
plt.title("Bar Plot of Votes")
plt.show()

File: 02-eda-1/bar-histogram.r
df <- read.csv("movies.csv")
df <- na.omit(df)
# Convert VOTES column from string to integer
df$VOTES <- as.integer(gsub(",", "", df$VOTES))

hist(df$VOTES, breaks = 10, xlab = "Votes", ylab = "Frequency", main = "Histogram of Votes")

# Reduce data to 10 records for better plot
df <- df[1:10, ]
barplot(df$VOTES, names.arg = df$MOVIES, xlab = "Movies", ylab = "Votes", main = "Bar Plot of Votes")

